Sources/SwiftBSON/BSONTimestamp.swift

```diff
@@ -91,6 +91,20 @@ public struct BSONTimestamp: BSONValue, Equatable, Hashable {
         ]
     }
 
+    /// Compares two `BSONTimestamp` instances as outlined by the `Comparable` protocol.
+    public static func < (lhs: BSONTimestamp, rhs: BSONTimestamp) -> Bool {
+        if lhs.timestamp != rhs.timestamp {
+            return lhs.timestamp < rhs.timestamp
+        } else { // equal timestamps, look at increment
+            return lhs.increment < rhs.increment
+        }
+    }
+
+    /// Checks two `BSONTimestamp` instances for equality as outlined by the `Comparable` protocol.
+    public static func == (lhs: BSONTimestamp, rhs: BSONTimestamp) -> Bool {
```

Comment:
We actually don't need to supply this custom implementation, since the auto generated one behaves the same way (this is what we were using before actually via the `Equatable` conformance).

We do need to keep the custom `Comparable` conformance in order to specify the precedence between timestamp and increment though.